ledger/participant-integration-api/src/main/scala/platform/apiserver/ApiServerConfig.scala

```diff
@@ -4,31 +4,62 @@
 package com.daml.platform.apiserver
 
 import com.daml.ledger.api.tls.TlsConfiguration
-import com.daml.lf.data.Ref
 import com.daml.platform.apiserver.SeedService.Seeding
-import com.daml.platform.configuration.{IndexConfiguration, InitialLedgerConfiguration}
-import com.daml.ports.Port
-import java.nio.file.Path
-import java.time.Duration
-
+import com.daml.platform.configuration.{
+  CommandConfiguration,
+  InitialLedgerConfiguration,
+  PartyConfiguration,
+}
+import com.daml.platform.services.time.TimeProviderType
+import com.daml.platform.store.DbSupport.{ConnectionPoolConfig, DbConfig}
 import com.daml.platform.usermanagement.UserManagementConfig
+import com.daml.ports.Port
 
-import scala.concurrent.duration.FiniteDuration
+import java.nio.file.Path
+import scala.concurrent.duration._
 
 case class ApiServerConfig(
-    participantId: Ref.ParticipantId,
-    port: Port,
-    address: Option[String], // This defaults to "localhost" when set to `None`.
-    jdbcUrl: String,
-    databaseConnectionPoolSize: Int,
-    databaseConnectionTimeout: FiniteDuration,
-    tlsConfig: Option[TlsConfiguration],
-    maxInboundMessageSize: Int,
-    initialLedgerConfiguration: Option[InitialLedgerConfiguration],
-    configurationLoadTimeout: Duration,
-    indexConfiguration: IndexConfiguration,
-    portFile: Option[Path],
-    seeding: Seeding,
-    managementServiceTimeout: Duration,
-    userManagementConfig: UserManagementConfig,
+    address: Option[String] =
+      ApiServerConfig.DefaultAddress, // This defaults to "localhost" when set to `None`.
+    apiStreamShutdownTimeout: Duration = ApiServerConfig.DefaultApiStreamShutdownTimeout,
```

Comment:
Duplicated with `IndexServiceConfig.apiStreamShutdownTimeout`

Reply:
I duplicated it for a reason. There is one in `ApiServerConfig` for the creation of `TimeServiceAuthorization` and another one in for `IndexServiceBuilder` to define `shutdownTimeout` in `Dispatcher`. 

As the name of parameter is the same - they configure different parameters for the different services. I did not want to push this param to `ParticipantConfig` as it would look like detached from the place where it is actually used. Which was one of the points for this refactoring - to push params to the places where they're used.

Reply:
If they configure different things, they should be named differently otherwise it's confusing, don't you think?

Reply:
it is a stream shutdown timeout for different components, do you want to configure global `apiStreamShutdownTimeout`?

If global - then `ParticipantConfig` would need to define it, and this is exactly what I wanted to avoid as to have "god object" with many parameters which are global as current `Config` in `main` branch.

Reply:
Original intention was to have one. We need the same behavior for all api streams (so for both cases). So in this case I guess pushing params to the right place: pushing it above.
In this case I feel like less configurability is better: it would be confusing to know that if you configure one, you should probably configure another in a different place with the same name.
Would it be possible to have one config instead of two?